### DISCUSSION TITLE
Support custom XML?

### DISCUSSION THREAD

**Discussion by uiosun** [WeblateOrg/weblate#6378] [Q&A]
Hi, I want to translate Age of Empires Online use Weblate, but the files struct is custom.

I read https://docs.weblate.org/en/latest/formats.html, but I think the xml format not include "Flat XML" or "Excel Open XML"(but the language .xml file can open from excel...)

Maybe I'm mistake, so please you can suggestion me. this is a real example by AOE Online language file:

```xml
﻿<?xml version="1.0" encoding="utf-8"?>
<stringtable xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns:xsd="http://www.w3.org/2001/XMLSchema" version="0" locstart="800000" locend="899999" loccurrent="802030">
	<language name="English">
		<string _locid="800000">Animal Hides</string>
		<string _locid="800001">Bronze Ingots</string>
		<string _locid="800002">Cedar Planks</string>
		<string _locid="800003">Copper Ingots</string>
		<string _locid="807038">Use: Builds a Persian-themed Trade Dock (decoration)</string>
		<string _locid="807039">Use: Builds a Babylonian-themed Trade Dock (decoration)</string>
		<string _locid="807040">Use: Builds a Norse-themed Trade Dock (decoration)</string>
		<string _locid="807041">Use: Builds a Roman-themed Trade Dock (decoration)</string>
		<string _locid="807042">Sacred Sheep Wool</string>
		<!-- ====================================================-->
		<!-- Celeste Sparta Stuff -->
		<string _locid="900001">Spartan Token</string>
	</language>
</stringtable>
```

ps: I'm a PHP/C# developer, if you think I can do something, please tell me.
I'd be absolutely delighted to learn something, and submit a PR.

**Comment by nijel** [comment]
I doubt that any of the existing XML formats fit for this. The preferred approach would be to add support for this to the translate-toolkit. Integrating into Weblate is then easy.

Basing on top of lisa classes (what does care of the XML processing) should be quite easy, take a look at https://github.com/translate/translate/blob/master/translate/storage/tmx.py for different XML format.

**Reply by uiosun** [reply]
I try use "Flat XML" importing the project, but panel tell me: need "Monolingual base language file" and "Template for new translations".

How to create or get that?

**Reply by nijel** [reply]
See https://docs.weblate.org/en/latest/formats.html#bimono and https://docs.weblate.org/en/latest/admin/projects.html#component-configuration
